conans/paths.py

```diff
@@ -45,7 +45,9 @@ def conan_expand_user(path):
         old_env = dict(os.environ)
         try:
             home = os.environ.get("HOME")
-            if home and not os.path.exists(home):
+            # Problematic cases of wrong HOME variable
```

Comment:
Please, write a comment about what and where the MSYSTEM/USERPROFILE. Are they variables set with MSYS console? Try to explain what it solves (title of the PR) in the code.